Makefile

```diff
@@ -81,8 +82,16 @@ $(SUBDIRS): force ## Execute default make target(make all) for the provided subd
 
 tests-privileged: ## Run Go tests including ones that require elevated privileges.
 	@$(ECHO_CHECK) running privileged tests...
-	PRIVILEGED_TESTS=true PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) -p 1 $(TEST_LDFLAGS) \
-		$(TESTPKGS) $(GOTEST_BASE) $(GOTEST_COVER_OPTS) | $(GOTEST_FORMATTER)
+	## We split tests into two parts: one that can be run in parallel
+	## and tests that cannot be run in parallel with other packages
+	## Two drawbacks of this approach:
+	## - if first set of tests fails, second one is not run
+	## - second part of tests does not report profile coverage.
+	## but it's better than flaky tests...
+	{ PRIVILEGED_TESTS=true PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) $(TEST_LDFLAGS) \
+		$(TESTPKGS) $(GOTEST_BASE) $(GOTEST_COVER_OPTS) \
+	&& PRIVILEGED_TESTS=true PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) $(TEST_LDFLAGS) \
+		$(UNPARALLELTESTPKGS) $(GOTEST_BASE) -json -covermode=count -p 1 --tags=unparallel; } | $(GOTEST_FORMATTER)
```

Comment:
Thanks for the note.
Regarding the change, probably if 1st set of tests fails then the 2nd one is not executed, therefore `tail -n+2 coverage2.out >> coverage.out` would fail because `coverage2.out` would not exist. Probably a `touch coverage2.out` before running the tests would solve it.

Reply:
> Regarding the change, probably if 1st set of tests fails then the 2nd one is not executed, therefore tail -n+2 coverage2.out >> coverage.out would fail because coverage2.out would not exist.

I don't think so, if the command fails in Makefile, Makefile just quits and never executes `tail -n+2 coverage2.out >> coverage.out`

Reply:
Yep you're right, my bad.